lib/vm_service_wrapper.dart

```diff
@@ -176,91 +232,122 @@ class VmServiceWrapper implements VmService {
+    return _trackFuture(_vmService.setExceptionPauseMode(isolateId, mode));
+  }
 
   @override
-  Future<Success> setFlag(String name, String value) =>
-      _trackFuture(_vmService.setFlag(name, value));
+  Future<Success> setFlag(String name, String value) {
+    return _trackFuture(_vmService.setFlag(name, value));
+  }
 
   @override
   Future<Success> setLibraryDebuggable(
-          String isolateId, String libraryId, bool isDebuggable) =>
-      _trackFuture(
-          _vmService.setLibraryDebuggable(isolateId, libraryId, isDebuggable));
+    String isolateId,
+    String libraryId,
+    bool isDebuggable,
+  ) {
+    return _trackFuture(
+        _vmService.setLibraryDebuggable(isolateId, libraryId, isDebuggable));
+  }
 
   @override
-  Future<Success> setName(String isolateId, String name) =>
-      _trackFuture(_vmService.setName(isolateId, name));
+  Future<Success> setName(String isolateId, String name) {
+    return _trackFuture(_vmService.setName(isolateId, name));
+  }
 
   @override
-  Future<Success> setVMName(String name) =>
-      _trackFuture(_vmService.setVMName(name));
+  Future<Success> setVMName(String name) {
+    return _trackFuture(_vmService.setVMName(name));
+  }
 
   @override
-  Future<Success> setVMTimelineFlags(List<String> recordedStreams) =>
-      _trackFuture(_vmService.setVMTimelineFlags(recordedStreams));
+  Future<Success> setVMTimelineFlags(List<String> recordedStreams) {
+    return _trackFuture(_vmService.setVMTimelineFlags(recordedStreams));
+  }
 
   @override
   Future<Success> streamCancel(String streamId) {
     _activeStreams.remove(streamId);
     return _trackFuture(_vmService.streamCancel(streamId));
   }
 
+  // We tweaked this method so that we do not try to listen to the same stream
+  // twice. This was causing an issue with the test environment and this change
+  // should not affect the run environment.
   @override
   Future<Success> streamListen(String streamId) {
-    Future<Success> future;
-    if (!_activeStreams.contains(streamId)) {
-      future = _trackFuture(_vmService.streamListen(streamId));
+    if (!_activeStreams.containsKey(streamId)) {
+      final Future<Success> future =
+          _trackFuture(_vmService.streamListen(streamId));
+      _activeStreams[streamId] = future;
+      return future;
+    } else {
+      return _activeStreams[streamId];
     }
-    _activeStreams.add(streamId);
-    return future;
   }
 
   Future<T> _trackFuture<T>(Future<T> future) {
     _activeFutures.add(future);
     future.whenComplete(() {
       _activeFutures.remove(future);
-      if (_activeFutures.isEmpty && !noPendingFutures.isCompleted) {
-        noPendingFutures.complete(true);
-      } else {
-        noPendingFutures = new Completer<bool>();
+      if (_activeFutures.isEmpty && !allFuturesCompleted.isCompleted) {
+        allFuturesCompleted.complete(true);
+      } else if (allFuturesCompleted.isCompleted) {
+        allFuturesCompleted = new Completer<bool>();
```

Comment:
My original advice was inaccurate . 
Remove the else case completely.
In reality what you need to do is at the top of this method write
```dart
if (allFuturesCompleted.isCompleted) {
 allFuturesCompleted = new Completer().
}
```

Reply:
you only want to be creating a new completer when the old completer was completed but you now know there is more work to do.

Reply:
Done.